config/base/derive/src/proxy.rs

```diff
@@ -35,86 +37,115 @@ pub fn impl_proxy(ast: StructWithFields) -> TokenStream {
     let set_field = ast.fields
         .iter()
         .map(|field| {
             let ty = &field.ty;
-            let l_value = &field.lvalue_write;
             let as_str_attr = field.has_as_str;
             let ident = &field.ident;
             let field_env = &field.env_str;
 
             let set_field = {
-                let inner_ty = field.has_option.then(|| get_inner_type("Option", ty)).unwrap_or(ty);
+                let inner_ty = if field.has_option {
+                    get_inner_type("Option", ty)
+                } else {
+                    abort!(ast, "This macro should only be used on `..Proxy` types")
+                };
                 let is_string = if let Type::Path(TypePath { path, .. }) = inner_ty {
                     path.is_ident("String")
                 } else {
                     false
                 };
+                let err_ty = quote! { ::iroha_config_base::derive::Error };
+                let err_variant = quote! { ::iroha_config_base::derive::Error::SerdeError };
                 let inner = if is_string {
                     quote! { Ok(var) }
                 } else if as_str_attr {
-                    quote! { serde_json::from_value(var.into()) }
+                    quote! { serde_json::from_value(var.into()).map_err(#err_variant) }
                 } else {
-                    quote! { serde_json::from_str(&var) }
+                    quote! { serde_json::from_str(&var).map_err(#err_variant) }
                 };
-                quote! {
-                    let inner: Result<#inner_ty, _> = #inner;
-                    #l_value = <#ty as From<#inner_ty>>::from(
-                        inner.map_err(|e| ::iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
-                    );
-                }
-            };
-
-            let inner_thing2 = if field.has_inner && field.has_option {
-                let inner_ty = get_inner_type("Option", ty);
-                let loadenv_trait = quote! { ::iroha_config_base::proxy::LoadFromEnv };
-                let err_variant = quote! { ::iroha_config_base::derive::Error::ProxyBuildError };
-                quote! {
-                    <#inner_ty as #loadenv_trait>::load_environment(#l_value.as_mut().ok_or(#err_variant(stringify!(#ident).to_owned()))?)?;
-                }
-            } else if field.has_inner {
-                quote! {
-                    #l_value.load_environment()?;
+                if field.has_inner {
+                    quote! {
+                        let #ident = if let Ok(var) = std::env::var(#field_env) {
+                            let inner: Result<#inner_ty, #err_ty> = #inner;
+                            inner.ok()
+                        } else { None };
+                        let inner_proxy = <#inner_ty as ::iroha_config_base::proxy::LoadFromEnv>::from_env();
+                        let #ident = if #ident.is_some() {
```

Comment:
While not `if let Some(...)`? instead of `unwrap`.

Reply:
A little bit of Sam's influence.